src/components/Deposit.tsx

```
@@ -0,0 +1,54 @@
+import useDeposit from "../hooks/useDeposit";
+
+export default function Deposit() {
+
+  const dialogRef = useRef<HTMLDialogElement>(null);
+  const [file, setFile] = useState<File | null>(null);
+  const { balance, network, account } = useWallet();
+  if (!network || !account.address) {
+    throw new Error('Network or account not found');
+  }
+	const { setDepositData, depositData, approve, deposit } = useDeposit(network, account.address);
+  const [isApproved, setIsApproved] = useState(false);
```

ESLint

[error] 'setIsApproved' is assigned a value but never used.

(@typescript-eslint/no-unused-vars)